lib/corm/model.rb

```diff
@@ -138,6 +140,43 @@ def self.count
       execute(statements['count']).first['count'].to_i
     end
 
+    ##
+    # Find by keys.
+    # This `find` methods wants to be as flexible as possible.
+    #
+    # Unless a block is given, it returns an `Enumerator`, otherwise it yields
+    # to the block an instance of the found Cassandra entries.
+    #
+    # If no keys as passed as parameter, the methods returns (an Enumerator for)
+    # all the results in the table.
+    #
+    # If one or more keys are given, the **order matters**; since this methods
+    # will craft a CSQL query, the first key **must** be the `partition key`,
+    # while the followings are intended as the `clustering keys`... that can be
+    # more than one.
+    #
+    # If the keys passed as parameter are more than the defined by the table the
+    # query is not valid, it cannot be executed and an error is raised.
+    def self.find(*key_values, &block)
+      return to_enum(:find, *key_values) unless block_given?
+
+      field_names = []
+      Array(key_values).each_with_index do |value, index|
+        field_names << "#{primary_key.flatten[index]} = ?"
+      end
+
+      statement_find_key = "find_#{Digest::MD5.hexdigest(field_names.inspect)}"
+
+      if statements[statement_find_key].nil?
+        statement = self.the_find_statement_for(key_values, field_names)
+        statements[statement_find_key] = session.prepare(statement)
+      end
+
+      execute(statements[statement_find_key], arguments: key_values).each do |cassandra_record_|
+        block.call(new(_cassandra_record: cassandra_record_))
+      end
+    end
+
     def self.get(relations)
       if statements['get'].nil?
         fields = primary_key.flatten.map { |key| "#{key} = ?" }.join ' AND '
```

Comment:
Instead of get all the primary keys as fields it's a matter to select the only ones passed via relations.